homeassistant/helpers/template_entity.py

```diff
@@ -597,3 +598,35 @@ def _render_templates(self, variables: dict[str, Any]) -> None:
                 "Error rendering %s template for %s: %s", key, self.entity_id, err
             )
             self._rendered = self._static_rendered
+
+
+class ManualTriggerEntity(TriggerBaseEntity):
+    """Template entity based on manual trigger data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: dict,
+    ) -> None:
+        """Initialize the entity."""
+        TriggerBaseEntity.__init__(self, hass, config)
+
+    @callback
+    def _process_manual_data(self, value: str | None = None) -> None:
+        """Process new data manually.
+
+        Implementing class should call this last in update method to render templates.
+        Ex: self._process_manual_data(payload)
+        """
+
+        self.async_write_ha_state()
+        this = None
+        if state := self.hass.states.get(self.entity_id):
+            this = state.as_dict()
+
+        run_variables: dict[str, Any] = {"value": value}
+        with contextlib.suppress(*JSON_DECODE_EXCEPTIONS):
```

Comment:
Why do we suppress JSON errors here?

Reply:
It's for testing if it's a `value_json` but if it's not then it's already stored in `value` as per line above and we should just continue.